packages/@ember/-internals/glimmer/lib/renderer.ts

```diff
@@ -95,27 +95,39 @@ class RootState {
       alwaysRevalidate: false,
     });
 
-    this.render = () => {
-      let layout = template.asLayout();
-      let handle = layout.compile();
-      let iterator = renderMain(
-        layout['compiler'].program,
-        env,
-        self,
-        dynamicScope,
-        builder(env, { element: parentElement, nextSibling: null }),
-        handle
-      );
-      let iteratorResult: IteratorResult<RenderResult>;
-      do {
-        iteratorResult = iterator.next();
-      } while (!iteratorResult.done);
-
-      let result = (this.result = iteratorResult.value);
-
-      // override .render function after initial render
-      this.render = () => result.rerender(options);
-    };
+    function handleError(fn: Function) {
```

Comment:
Can you place this function into an `if (DEBUG)` statement? It looks like the debug stack is only included in development builds, and the `DEBUG` environment variable actually causes all of that code to be completely removed from production builds. This would prevent us from adding additional overhead in production, but allow us to keep the full benefits in dev.